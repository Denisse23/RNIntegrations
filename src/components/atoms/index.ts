export { default as CustomText } from './CustomText/CustomText';
export { default as Button } from './Button/Button';
export { default as CustomInput } from './CustomInput/CustomInput';
export { type CustomInputProps } from './CustomInput/CustomInput';
