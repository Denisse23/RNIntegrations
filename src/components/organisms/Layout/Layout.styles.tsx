import styled from '@emotion/native';
import { Container } from '@styles/SharedStyles';

export const StyledSafeAreView = styled.SafeAreaView(() => ({
  flex: 1,
}));

export const StyledContainer = styled(Container)(({ theme }) => ({
  padding: theme.spacings.xxl,
}));
