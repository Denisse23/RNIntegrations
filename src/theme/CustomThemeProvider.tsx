import React from 'react';
import { useColorScheme } from 'react-native';
import { ThemeProvider } from '@emotion/react';
import { ColorScheme, DarkTheme, LightTheme } from './Themes';

const CustomThemeProvider = ({ children }: { children: React.ReactNode }) => {
  const scheme = useColorScheme();

  return (
    <ThemeProvider theme={scheme === ColorScheme.DARK ? DarkTheme : LightTheme}>
      {children}
    </ThemeProvider>
  );
};

export default CustomThemeProvider;
