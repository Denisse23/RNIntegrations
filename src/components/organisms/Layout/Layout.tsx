import React from 'react';
import { ScrollView, ViewProps } from 'react-native';
import { StyledContainer, StyledSafeAreView } from './Layout.styles';

export interface LayoutProps {
  scroll?: boolean;
  children?: React.ReactNode;
  testID?: ViewProps['testID'];
}

const Layout = ({ scroll = true, testID, children }: LayoutProps) => {
  const getContainer = () => {
    return <StyledContainer>{children}</StyledContainer>;
  };
  return (
    <StyledSafeAreView testID={testID} accessibilityLabel={testID}>
      {scroll ? (
        <ScrollView showsVerticalScrollIndicator={false}>
          {getContainer()}
        </ScrollView>
      ) : (
        getContainer()
      )}
    </StyledSafeAreView>
  );
};

export default Layout;
