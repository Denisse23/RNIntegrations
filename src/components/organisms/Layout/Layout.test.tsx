import React from 'react';
import renderer from 'react-test-renderer';
import Layout from './Layout';
import { CustomText } from '@components/atoms';
import CustomThemeProvider from '@theme/CustomThemeProvider';

describe('Layout', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(
      <CustomThemeProvider>
        <Layout>
          <CustomText>Test</CustomText>
        </Layout>
      </CustomThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
