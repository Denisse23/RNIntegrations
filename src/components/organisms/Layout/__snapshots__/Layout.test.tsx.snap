// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match the snapshot 1`] = `
<RCTSafeAreaView
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Array [
              Object {
                "padding": 37.5,
              },
              undefined,
            ],
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#353434",
                "fontSize": 28.125,
                "fontWeight": "400",
              },
              undefined,
            ]
          }
        >
          Test
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
